### PR TITLE
[weather.noaa] 2.0.3

### PR DESCRIPTION
**Pull request by randallspicher**
### Description
Weather plugin that pulls data from the weather.gov (noaa) public api
### Checklist:
- [X ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X ] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X ] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: weather.noaa/addon.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	<requires>
+		<import addon="xbmc.python" version="3.0.0"/>
+		<import addon="script.module.dateutil" version="2.7.3"/>
+		<import addon="script.module.future" version="0.17.1"/>
+	</requires>
+	<extension point="xbmc.python.weather" library="default.py"/>
+	<extension point="xbmc.addon.metadata">
+		<summary lang="en_GB">Weather forecast from weather.gov (NOAA)</summary>
+		<description lang="en_GB">Weather forecast provided by NOAA (http://weather.gov/)</description>
+		<language>en_US</language>
```
Comment (enen92) [inline weather.noaa/addon.xml:12]: this only applies to media. Please remove the language element altogether

File: weather.noaa/addon.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		<platform>all</platform>
+		<license>GNU GENERAL PUBLIC LICENSE Version 2</license>
```
Comment (enen92) [inline weather.noaa/addon.xml:14]:
```suggestion
		<license>GPL-2.0-only</license>
```

or 

```suggestion
		<license>GPL-2.0-or-later</license>
```

(SPDX identifiers)

File: weather.noaa/addon.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		<forum></forum>
```
Comment (enen92) [inline weather.noaa/addon.xml:15]: drop if empty

File: weather.noaa/addon.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		<website>https://github.com/randallspicher/weather.noaa</website>
+		<email></email>
```
Comment (enen92) [inline weather.noaa/addon.xml:17]: drop if empty

File: weather.noaa/default.py [hunk 5]
```diff
@@ -0,0 +1,776 @@
...
+from resources.lib.utils import WEATHER_CODES, FORECAST, FEELS_LIKE, SPEED, WIND_DIR, SPEEDUNIT, zip_x
+from resources.lib.utils import get_url_JSON, decode_utf8  #, encode_utf8
+from resources.lib.utils import get_month, get_timestamp, get_weekday, get_time
+
+
+
+ADDON           = xbmcaddon.Addon()
+ADDONNAME       = ADDON.getAddonInfo('name')
+ADDONID         = ADDON.getAddonInfo('id')
+CWD             = decode_utf8(ADDON.getAddonInfo('path'))
```
Comment (enen92) [inline weather.noaa/default.py:22]: decode_utf8 shouldn't be needed here

File: weather.noaa/default.py [hunk 6]
```diff
@@ -0,0 +1,776 @@
...
+ADDONVERSION    = ADDON.getAddonInfo('version')
+LANGUAGE        = ADDON.getLocalizedString
+RESOURCE        = decode_utf8(xbmc.translatePath(os.path.join( CWD, 'resources', 'lib' ).encode("utf-8") ))
```
Comment (enen92) [inline weather.noaa/default.py:25]: same

File: weather.noaa/default.py [hunk 7]
```diff
@@ -0,0 +1,776 @@
...
+PROFILE         = decode_utf8(xbmc.translatePath(ADDON.getAddonInfo('profile')))
```
Comment (enen92) [inline weather.noaa/default.py:26]: same

File: weather.noaa/default.py [hunk 8]
```diff
@@ -0,0 +1,776 @@
...
+
+sys.path.append(RESOURCE)
```
Comment (enen92) [inline weather.noaa/default.py:28]: sorry, no `sys.path.append`. Please use relative module imports:

`from resources.lib import mymodule`

File: weather.noaa/default.py (hunk 6 above)
Comment (enen92) [inline weather.noaa/default.py:25]:
```suggestion
RESOURCE        = xbmcvfs.translatePath(os.path.join( CWD, 'resources', 'lib' )
```

File: weather.noaa/default.py (hunk 7 above)
Comment (enen92) [inline weather.noaa/default.py:26]:
```suggestion
PROFILE         =  xbmcvfs.translatePath(ADDON.getAddonInfo('profile'))
```

File: weather.noaa/default.py [hunk 9]
```diff
@@ -0,0 +1,776 @@
...
+
+
+WEATHER_ICON	= decode_utf8(xbmc.translatePath('%s.png'))
```
Comment (enen92) [inline weather.noaa/default.py:31]:
```suggestion
WEATHER_ICON	= xbmcvfs.translatePath('%s.png')
```

File: weather.noaa/default.py [hunk 10]
```diff
@@ -0,0 +1,776 @@
...
+		xbmc.log('%s' % alerts,level=xbmc.LOGINFO)
+		return
+		
+	if 'features' in alerts and alerts['features']:
+		data=alerts['features']
+		#xbmc.log('data: %s' % data,level=xbmc.LOGINFO)
+		set_property('Alerts.IsFetched'	, 'true')
+	else:
+		clear_property('Alerts.IsFetched')
+		xbmc.log('No current weather alerts from  %s' % url,level=xbmc.LOGINFO)
```
Comment (enen92) [inline weather.noaa/default.py:573]:
```suggestion
		xbmc.log('No current weather alerts from  %s' % url,level=xbmc.LOGDEBUG)
```

File: weather.noaa/default.py [hunk 11]
```diff
@@ -0,0 +1,776 @@
...
+	url="https://api.weather.gov/alerts/active/zone/%s" %a_zone	
+	alerts=get_url_JSON(url)
+	#xbmc.log('current data: %s' % current_data,level=xbmc.LOGINFO)
+	# if we have a valid response then clear our current alerts
+	if alerts and alerts != '' and 'features' in alerts:
+		for count in range (1, 10):
+			clear_property('Alerts.%i.event' % (count))	
+	else:
+		xbmc.log('failed to get proper alert response %s' % url,level=xbmc.LOGERROR)
+		xbmc.log('%s' % alerts,level=xbmc.LOGINFO)
```
Comment (enen92) [inline weather.noaa/default.py:564]:
```suggestion
		xbmc.log('%s' % alerts,level=xbmc.LOGDEBUG)
```

File: weather.noaa/default.py [hunk 12]
```diff
@@ -0,0 +1,776 @@
...
+		if "forecast.weather.gov" == sourcePref:
+			fetchAltDaily(num)
+		else:
+			fetchCurrent(num)
+			#currentforecast(num)
+			fetchDaily(num)
+		fetchHourly(num)
+		##Station=ADDON.getSetting('Location%scwa' % num)
+		Station=ADDON.getSetting('Location%sradarStation' % num)
+
```
Comment (enen92) [inline weather.noaa/default.py:721]: would be nice if you could cleanup your commented code

File: weather.noaa/default.py [hunk 13]
```diff
@@ -0,0 +1,776 @@
...
+				clear_property('Map.%i.Area' % (mcount))
+				clear_property('Map.%i.Heading' % (mcount))
+				clear_property('Map.%i.Layer' % (mcount))
+	else:
+		log('no location provided')
+		clear()
+
+log('finished')
+
+
```
Comment (enen92) [inline weather.noaa/default.py:776]: empty `__init__.py` files shouldn't be needed anymore with python3

File: weather.noaa/resources/language/resource.language.en_GB/strings.po [hunk 14]
```diff
@@ -0,0 +1,502 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline weather.noaa/resources/language/resource.language.en_GB/strings.po:1]: language/resource.language.en_gb/strings.po not language/resource.language.en_GB/strings.po  (consistency with other addons)

File: weather.noaa/resources/lib/utils.py [hunk 15]
```diff
@@ -0,0 +1,532 @@
+# -*- coding: utf-8 -*-
```
Comment (enen92) [inline weather.noaa/resources/lib/utils.py:1]: dummy `__init__.py` files shouldn't be needed anymore

File: weather.noaa/resources/settings.xml [hunk 16]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (enen92) [inline weather.noaa/resources/settings.xml:1]: would be nice if you could migrate this to the new settings format at some point. See: https://kodi.wiki/view/Add-on_settings_conversion

Automated addon: https://gitlab.com/ronie/script.settings-convert

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments

File: weather.noaa/default.py (hunk 5 above)
Comment (randallspicher) [reply weather.noaa/default.py:22]: It breaks in Leia if the path isn't decoded to utf8.  (the decode_utf8() is a function to check the python version to determine whether-or-not to decode the string).  I'm not sure how to handle it without running that check.

**Comment by enen92** [comment]
ping @randallspicher

**Comment by randallspicher** [comment]
will open new pull request for the updated version of the plugin

File: weather.noaa/resources/.gitignore [hunk 17]
```diff
@@ -0,0 +1 @@
+/settings.bak.xml
```
Comment (enen92) [inline weather.noaa/resources/.gitignore:1]: Sorry this file has to be removed from the submission

**Review by enen92 — CHANGES_REQUESTED** [review]
One minor (gitignore, good to go afterwards)

File: weather.noaa/resources/.gitignore (hunk 17 above)
Comment (randallspicher) [reply weather.noaa/resources/.gitignore]: This should be removed now.

**Review by enen92 — APPROVED** [review]

**Pull request merged**